config/new.go

```diff
@@ -96,6 +96,13 @@ func NewHCPConfig(opts ...HCPConfigOption) (HCPConfig, error) {
 		}
 	}
 
+	// fail out with a typed error if invoker specified WithoutBrowserLogin
+	if config.noBrowserLogin {
```

Comment:
Instead of reading this boolean here, I'd rather add this to the block of conditional logic below, where we switch between client credentials and user (OAuth2) credentials. 

So this `else`: https://github.com/hashicorp/hcp-sdk-go/blob/9e36fe8e940c59600184ea0ccd1e6dac59e6e0d6/config/new.go#L125
can become `} else if config.oauth2Config != oauth2.Config{} { `.
And then after that, in the final `else`, we can throw the `ErrorNoValidAuthFound`.